README.md

```diff
@@ -99,36 +99,93 @@ book].
 
 #### Android
 
-This library makes use of the [ndk-rs](https://github.com/rust-windowing/android-ndk-rs) crates, refer to that repo for more documentation.
+The Android backend builds on (and exposes types from) the [`ndk`](https://docs.rs/ndk/0.7.0/ndk/) crate.
 
-The `ndk-glue` version needs to match the version used by `winit`. Otherwise, the application will not start correctly as `ndk-glue`'s internal `NativeActivity` static is not the same due to version mismatch.
+Native Android applications need some form of "glue" crate that is responsible
+for defining the main entry point for your Rust application as well as tracking
+various life-cycle events and synchronizing with the main JVM thread.
 
-`winit` compatibility table with `ndk-glue`:
+Winit uses the [android-activity](https://github.com/rib/android-activity) as a
+glue crate (prior to `0.28` it used
+[ndk-glue](https://github.com/rust-windowing/android-ndk-rs/tree/master/ndk-glue).
 
-| winit |       ndk-glue       |
-| :---: | :------------------: |
-| 0.24  | `ndk-glue = "0.2.0"` |
-| 0.25  | `ndk-glue = "0.3.0"` |
-| 0.26  | `ndk-glue = "0.5.0"` |
-| 0.27  | `ndk-glue = "0.7.0"` |
+The version of the glue crate that your application depends on _must_ match the
+version that Winit depends on because the glue crate is responsible for your
+application's main entrypoint. If Cargo resolves multiple versions they will
+clash.
+
+`winit` glue compatibility table:
+
+| winit |       ndk-glue               |
+| :---: | :--------------------------: |
+| 0.28  | `android-activity = "0.4"`   |
+| 0.27  | `ndk-glue = "0.7"`           |
+| 0.26  | `ndk-glue = "0.5"`           |
+| 0.25  | `ndk-glue = "0.3"`           |
+| 0.24  | `ndk-glue = "0.2"`           |
+
+The recommended way to avoid a conflict with the glue version is to avoid explicitly
+depending on the `android-activity` crate, and instead consume the API that
+is re-exported by Winit under `winit::platform::android::activity::*`
 
 Running on an Android device needs a dynamic system library, add this to Cargo.toml:
 
 ```toml
-[[example]]
-name = "request_redraw_threaded"
+[lib]
+name = "main"
 crate-type = ["cdylib"]
 ```
 
-And add this to the example file to add the native activity glue:
+All Android applications are based on an `Activity` subclass and the
+`android-activity` crate is designed to support different choices for this base
+class. You application must specify the base class it needs via a feature flag:
+
+| Base Class       | Feature Flag      |  Notes  |
+| :--------------: | :---------------: | :-----: |
+| `NativeActivity` | `android-native-activity` | Built-in to Android - making it possible to build some tests/demos without needing to compile any JVM code. Can give a false sense of convenience because it's often not really possible to avoid needing a build system that can compile some JVM code, to at least subclass `NativeActivity` |
```

Comment:
I've taken a pass at tweaking the wording here.

I've tried to remove the language that implies it's only "tests/demos" that can use NativeActivity or assuming anything about the false sense of convenience it has.

The general intent here was to try and highlight a few of the pros/cons of each option that can help someone make a choice.

I didn't say anything about lack of extensibility, from the pov that developers can subclass `NativeActivity` if they need to (so GameActivity / NativeActivity are both similarly extensible) - but did highlight that it may be necessary to subclass `NativeActivity` to access some platform features.

So although it's notably possible to use `NativeActivity` without compiling any Java/Kotlin code, it's hopefully also clear that it may sometimes be necessary to subclass `NativeActivity`.

I'm not super happy with how these notes are currently packed in this table though, since it squashes the feature name and it's awkward to format the notes. Maybe the notes should be split out into some pros/cons afterwards - not quite sure atm.